src/codeswitch/data_generation/generate_prompt_template.py

```diff
@@ -0,0 +1,143 @@
+import pandas as pd
+
+# Load OpenAI API key from environment variable
+# openai.api_key = os.getenv("open_ai_key")
+
+
+# Function to generate code-switched prompts using the fine-tuned model
+def generate_code_switched_prompts(contexts, num_instances=10):
+    prompts = []
+    for context in contexts:
+        prompt = generate_prompt(context, num_instances)
+        prompts.extend(prompt)
+    return prompts
+
+
+# Function to generate prompts for each context
+def generate_prompt(context, num_instances):
+    template = get_template(context)
+    prompts = []
+    for i in range(num_instances):
+        prompt = f"{template}\n\n[English-Spanish Code-Switched Utterances {i+1}]:"
+        prompt_label = f"[{context}]"
+        prompts.append(
+            (prompt_label, prompt, context)
+        )  # Modified to include the prompt label and domain
+    return prompts
+
+
+# Function to get template for a given context
+def get_template(context):
+    templates = {
+        "Business Meeting": """
+        [Business Meeting]:
```

Comment:
The prompt can be improved. 

1. You should mention the overall goal of the task at the beginning, which is to generate code switch sentences. This is never mentioned in the current prompt.

2. Instead of writing like "[Business Meeting]". You should write something in natural language like "Generate some code-switch sentences that are likely to appear in a conversation of business meetings." It is currently unclear on the purpose of the bracketed phrase.


3. Instead of just listing requirements, you should mention they are requirements, such as
"
Follow the guidelines closely:
        - Use a mix of formal and informal language to reflect different communication settings.
        - Add code-switching examples in other languages depending on the target audience and purpose of the dataset.
"

4. Some parts of the template are repeated. Instead of writing it as a string everytime, simply concat them to the end of the changing string (Unless you have reason to believe they might not be the same in the future). For example:

```
prompt += """
        [Insert an English sentence with a noun phrase that is likely to be code-switched to Spanish.]
        [Insert an English sentence with an action verb phrase that is likely to be code-switched to Spanish.]
        [Insert an English sentence with an object phrase that is likely to be code-switched to Spanish.]
        [Insert a Spanish sentence with a noun phrase that is likely to be code-switched to English.]
        [Insert a Spanish sentence with an action verb phrase that is likely to be code-switched to English.]
        [Insert a Spanish sentence with an object phrase that is likely to be code-switched to English.]"""
```

5. The wording of "insert" is awkward in this case, as there is nothing to insert into. You can say "Give me an english sentence....",  or give it something to insert into, like a json list:

```
Give me a json of the following structure:

{
    "sentences":
      [
        "[Insert an English sentence with a noun phrase that is likely to be code-switched to Spanish.]",
        "[Insert an English sentence with an action verb phrase that is likely to be code-switched to Spanish.]",
        "[Insert an English sentence with an object phrase that is likely to be code-switched to Spanish.]",
      ]
}
``` 




Dont simply follow each of my suggestions blindly, they are not directly compatible Figure out a way to combine all of them.

The general guideline to writing prompt is, if you give the prompt to a human, how likely will the human understand what you're asking? The easier to understand the better it is.